doc/source/rllib-concepts.rst

```diff
@@ -581,12 +581,141 @@ Here is an example of creating a set of rollout workers and using them gather ex
         for w in workers.remote_workers():
             w.set_weights.remote(weights)
 
-Execution Plan
---------------
+Execution Plans
+---------------
 
-.. note::
+Execution plans let you easily express the execution of an RL algorithm as a sequence of steps that
+occur either sequentially in the learner, or in parallel across many actors.
+Under the hood, RLlib *translates* these plans into ``ray.get()`` and ``ray.wait()`` operations over Ray actors,
+so you easily write high-performance algorithms without needing to manage individual low-level Ray actor calls.
 
-    Policy optimizers have been replaced by the execution plan API. This documentation will be updated in the future.
+Execution plans represent the **dataflow of the RL training job**. For example, the A2C algorithm can be thought
+of a sequence of repeating steps, or *dataflow*, of:
+
+ 1. ``ParallelRollouts``: Generate experiences from many envs in parallel using rollout workers.
+ 2. ``ConcatBatches``: The experiences are concatenated into one batch for training.
+ 3. ``TrainOneStep``: Take a gradient step with respect to the policy loss, and update the worker weights.
+
+In code, this dataflow can be expressed as the following execution plan:
+
+.. code-block:: python
+
+    def execution_plan(workers: List[RolloutWorker], config: TrainerConfigDict):
+        # type: LocalIterator[SampleBatchType]
+        rollouts = ParallelRollouts(workers, mode="bulk_sync")
+
+        # type: LocalIterator[(SampleBatchType, List[LearnerStatsDict])]
+        train_op = rollouts \
+            .combine(ConcatBatches(
+                min_batch_size=config["train_batch_size"])) \
+            .for_each(TrainOneStep(workers))
```

Comment:
It's a little confusing what for_each actually does.
- for_each worker?
or
- for_each `next` call of the iterator.